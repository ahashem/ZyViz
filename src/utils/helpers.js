/**
 * Provide shortened version (3 letters) of the provided Day of Week
 * @param weekDay
 * @return {string} shortened weekday
 */
const shortenWeekDay = (weekDay) => {
  const weekDays = {
    'Mondays': 'Mon',
    'Monday': 'Mon',
    'Tuesdays': 'Tue',
    'Tuesday': 'Tue',
    'Wednesdays': 'Wed',
    'Wednesday': 'Wed',
    'Thursdays': 'Thu',
    'Thursday': 'Thu',
    'Fridays': 'Fri',
    'Friday': 'Fri',
    'Saturdays': 'Sat',
    'Saturday': 'Sat',
    'Sundays': 'Sun',
    'Sunday': 'Sun',
  };

  return weekDays[`${weekDay}`];
};

/**
 * Format currency with USD
 * @param amount
 * @return {string}
 */
const currencyTickFormat = (amount) => (((amount / 1000) >= 1) ? `$${(amount / 1000)}K` : `$${amount}`);

export { shortenWeekDay, currencyTickFormat };
