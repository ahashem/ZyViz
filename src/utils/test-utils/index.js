import crossfilter from '../../components/crossfilter/crossfilter';
import NormalizeData from '../../views/CrossVisualizer/visualizer-data-structure/data-normalizer';
import mockOrdersJSON from './mock-test-orders';
import GenerateDimensions from '../../views/CrossVisualizer/visualizer-data-structure/dimensions-manager';

const mockOrdersCrossfilter = crossfilter(NormalizeData(mockOrdersJSON));
const mockOrdersDimensions = GenerateDimensions(mockOrdersCrossfilter);

export { mockOrdersJSON as jsonMock, mockOrdersCrossfilter as crossFilterMock, mockOrdersDimensions as dimensionsMock };
