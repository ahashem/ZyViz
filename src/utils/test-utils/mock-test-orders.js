const mockOrders = [
  {
    'userid': '69b8810c-12ef-4107-a282-03cddb46f1ad',
    'orderid': 1,
    'orderAmount': '$89.96',
    'orderdate': '2015-07-09T13:53:06+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch D',
    'deliveryArea': 'Al Bidae'
  },
  {
    'userid': 'f1d3caaf-be09-43d8-a7d9-5fa3ecb530fa',
    'orderid': 2,
    'orderAmount': '$36.51',
    'orderdate': '2014-07-24T14:33:52+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch D',
    'deliveryArea': 'Saad Al Abdullah'
  },
  {
    'userid': 'c94a020b-63d8-42bb-af31-4af88a52a383',
    'orderid': 3,
    'orderAmount': '$99.61',
    'orderdate': '2017-04-13T18:55:54+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch C',
    'deliveryArea': 'Qibla'
  },
  {
    'userid': 'f2cc211e-f835-4c87-b0e4-ef1b048f5d21',
    'orderid': 4,
    'orderAmount': '$20.85',
    'orderdate': '2017-09-30T01:37:22+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch A',
    'deliveryArea': 'Doha'
  },
  {
    'userid': '96877167-873b-4b49-aa5f-88cd59b56929',
    'orderid': 5,
    'orderAmount': '$83.02',
    'orderdate': '2015-12-16T23:19:31+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch F',
    'deliveryArea': 'Faiha'
  },
  {
    'userid': 'e1eeb0d3-4eef-48a1-9bf3-21a480223112',
    'orderid': 6,
    'orderAmount': '$74.36',
    'orderdate': '2014-06-26T08:03:06+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch E',
    'deliveryArea': 'Nuzha'
  },
  {
    'userid': '1f02b30c-6ab1-44ec-81be-572caab100e4',
    'orderid': 7,
    'orderAmount': '$34.84',
    'orderdate': '2017-01-01T09:24:30+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch D',
    'deliveryArea': 'Fahaheel'
  },
  {
    'userid': 'd55fa20a-7d00-4822-b6fa-f0f5ab1ec4ce',
    'orderid': 8,
    'orderAmount': '$91.43',
    'orderdate': '2017-05-31T10:53:02+00:00',
    'paymentMethod': 'KNET',
    'branch': 'Branch A',
    'deliveryArea': 'Al Qusour'
  },
  {
    'userid': '2e03c4eb-50fc-47a4-bf7f-547a55ddf278',
    'orderid': 9,
    'orderAmount': '$67.52',
    'orderdate': '2015-12-06T00:16:29+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch E',
    'deliveryArea': 'Abu Halifa'
  },
  {
    'userid': '6379f0a8-5f36-489b-b154-ac14360626cb',
    'orderid': 10,
    'orderAmount': '$30.51',
    'orderdate': '2017-10-24T09:33:48+00:00',
    'paymentMethod': 'KNET',
    'branch': 'Branch B',
    'deliveryArea': 'Qasser'
  },
  {
    'userid': '20d7304b-2eca-47e5-999c-b21b0af9e5b6',
    'orderid': 11,
    'orderAmount': '$83.43',
    'orderdate': '2017-02-01T17:39:54+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch B',
    'deliveryArea': 'Abdullah Al Salem'
  },
  {
    'userid': 'e17782e4-f4fb-4cf7-8ff5-89bd7e27df4c',
    'orderid': 12,
    'orderAmount': '$30.49',
    'orderdate': '2018-06-23T14:10:36+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch D',
    'deliveryArea': 'Qibla'
  },
  {
    'userid': 'b1689a89-3c1b-46e4-863d-e825a48b62cf',
    'orderid': 13,
    'orderAmount': '$7.64',
    'orderdate': '2018-06-02T01:20:40+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch B',
    'deliveryArea': 'Rabiya'
  },
  {
    'userid': '533de73c-053d-4c8d-b023-5a658ab387eb',
    'orderid': 14,
    'orderAmount': '$34.87',
    'orderdate': '2018-05-22T14:02:06+00:00',
    'paymentMethod': 'Cash',
    'branch': 'Branch F',
    'deliveryArea': 'Mirqab'
  },
  {
    'userid': '6a2edc6d-d74b-47b6-b554-f97c3ec5a391',
    'orderid': 15,
    'orderAmount': '$36.35',
    'orderdate': '2018-06-21T16:25:09+00:00',
    'paymentMethod': 'KNET',
    'branch': 'Branch F',
    'deliveryArea': 'Nahda'
  },
  {
    'userid': '5b317119-cda9-416f-923e-cb98ed5ea4a6',
    'orderid': 16,
    'orderAmount': '$32.93',
    'orderdate': '2016-02-24T13:05:42+00:00',
    'paymentMethod': 'KNET',
    'branch': 'Branch A',
    'deliveryArea': 'Reggai'
  },
  {
    'userid': '4f26a8e0-3199-4d08-b8bb-d0a5675e8972',
    'orderid': 17,
    'orderAmount': '$4.46',
    'orderdate': '2017-08-05T21:36:56+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch D',
    'deliveryArea': 'Jaber Al Ali'
  },
  {
    'userid': 'a3d2a038-21e1-40e3-8fe6-ef0c4a1a8d5a',
    'orderid': 18,
    'orderAmount': '$78.78',
    'orderdate': '2016-08-10T16:56:58+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch F',
    'deliveryArea': 'Zahra'
  },
  {
    'userid': 'f6b30b19-a30a-40ae-8013-39d7c3c7b3c4',
    'orderid': 19,
    'orderAmount': '$85.38',
    'orderdate': '2017-07-30T14:23:52+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch A',
    'deliveryArea': 'Hadiya'
  },
  {
    'userid': '9b7eb604-3fd4-4360-9ff2-d5641570aed1',
    'orderid': 1233,
    'orderAmount': '$92.92',
    'orderdate': '2016-09-17T14:33:59+00:00',
    'paymentMethod': 'KNET',
    'branch': 'Branch C',
    'deliveryArea': 'Rai'
  },
  {
    'userid': 'ef48052d-581f-4911-953c-156936200e14',
    'orderid': 1234,
    'orderAmount': '$45.30',
    'orderdate': '2017-08-31T08:55:50+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch F',
    'deliveryArea': 'Qortuba'
  },
  {
    'userid': '76fa01fd-d9ab-495e-afad-fa12b228413b',
    'orderid': 1235,
    'orderAmount': '$83.29',
    'orderdate': '2015-12-25T14:35:04+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch F',
    'deliveryArea': 'Rai'
  },
  {
    'userid': 'e0025bee-b63b-44e7-b286-366649161082',
    'orderid': 1236,
    'orderAmount': '$1.84',
    'orderdate': '2014-12-04T14:25:12+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch B',
    'deliveryArea': 'Abu Al Hasaniya'
  },
  {
    'userid': '840812f4-0d22-4680-9554-6242ca4dc22f',
    'orderid': 1237,
    'orderAmount': '$34.16',
    'orderdate': '2017-10-05T10:35:00+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch D',
    'deliveryArea': 'Zahra'
  },
  {
    'userid': '17929585-a2f5-43bf-807f-cc78c64ce8b3',
    'orderid': 1238,
    'orderAmount': '$57.89',
    'orderdate': '2015-05-04T01:54:51+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch C',
    'deliveryArea': 'Sheikh Saad Aviation Terminal'
  },
  {
    'userid': '0c4bf56f-25da-4081-8e5e-c07d61d22db4',
    'orderid': 1239,
    'orderAmount': '$14.34',
    'orderdate': '2016-02-10T09:29:28+00:00',
    'paymentMethod': 'KNET',
    'branch': 'Branch F',
    'deliveryArea': 'Reggai'
  },
  {
    'userid': 'cf0ad774-0096-464f-ac00-3cbe37578904',
    'orderid': 1240,
    'orderAmount': '$78.24',
    'orderdate': '2018-03-31T04:01:29+00:00',
    'paymentMethod': 'CreditCard',
    'branch': 'Branch E',
    'deliveryArea': 'Jaber Al Ahmad'
  },
];

export default mockOrders;
