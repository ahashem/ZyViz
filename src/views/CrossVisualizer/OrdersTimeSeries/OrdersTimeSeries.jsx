import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FlexGridRow from '../../../components/flex-wrapper/FlexGridRow';
import TimeSeries from '../../../components/TimeSeries/TimeSeries';

/**
 * @component OrdersTimeSeries
 */
class OrdersTimeSeries extends Component {
  static propTypes = {
    orders: PropTypes.shape({}),
    dimension: PropTypes.shape({}),
  };

  static defaultProps = {
    orders: {},
    dimension: {},
  };

  /**
   * Receive a crossfilter dimensions, create groups to use for charts
   * @param {Object} dimensions
   * @return {Object<Array<crossfilter.Grouping>>}
   */
  prepareOrdersCountData = (dimensions) => {
    // Serialize and group data using crossfilter to be
    // for-example: {key: 'paymentMethod', value: 20}

    // Orders by Branch
    const ordersByDate = dimensions.ordersByDate;
    const ordersByDateDataGroup = ordersByDate.group().reduceCount().all();

    const ordersByDateData = ordersByDateDataGroup.map(data => {
      return {
        x: new Date(data.key),
        y: isNaN(data.value) ? 0 : data.value
      };
    });
    const totalSaleByDateDataGroup = ordersByDate.group().reduceSum(order => order.orderAmount).all();
    const totalSaleByDateData = totalSaleByDateDataGroup.map(data => {
      return {
        x: (data.key instanceof Date) ? data.key : new Date(Date.parse(data.key)),
        y: isNaN(data.value) ? 0 : data.value
      };
    });

    return {
      ordersByDateData,
      totalSaleByDateData,
    };

  };

  /**
   * Uplift filter triggers
   * @param {*|string} dimension
   * @param {Number} selectedKey
   * @param {Boolean} deselected
   */
  crossFilterSelected = (dimension, selectedKey, deselected = false) => {
    const { onFilter } = this.props;
    // console.log('selectedData', selectedKey);
    onFilter(dimension, selectedKey, deselected);
  };


  render() {
    const { dimensions } = this.props;
    const ordersCrossed = dimensions ? this.prepareOrdersCountData(dimensions) : null;

    return (
      <div>
        <FlexGridRow justify="space-around" gutter={48}>

          {ordersCrossed && (
            <TimeSeries
              data={ordersCrossed.ordersByDateData}
            />)}

        </FlexGridRow>
        <FlexGridRow justify="space-around" gutter={48}>

          {ordersCrossed && (
            <TimeSeries
              data={ordersCrossed.totalSaleByDateData}
            />)}

        </FlexGridRow>
      </div>
    );
  }
}

export default OrdersTimeSeries;
