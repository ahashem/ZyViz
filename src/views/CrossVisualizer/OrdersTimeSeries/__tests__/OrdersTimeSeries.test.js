import React from 'react';
import { shallow } from 'enzyme';

import OrdersTimeSeries from '../OrdersTimeSeries';
import { crossFilterMock, dimensionsMock } from '../../../../utils/test-utils';

describe('<OrdersTimeSeries /> Component', () => {

  test('should render children and match snapshot', () => {
    const orders = crossFilterMock;
    const dimensions = dimensionsMock;
    const onFilter = jest.fn();

    const rendered = shallow(<OrdersTimeSeries
      orders={orders}
      dimensions={dimensions}
      onFilter={onFilter}
    />);

    expect(rendered).toMatchSnapshot();
  });

});
