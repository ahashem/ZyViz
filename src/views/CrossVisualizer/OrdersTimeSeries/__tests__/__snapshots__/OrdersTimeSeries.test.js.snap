// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersTimeSeries /> Component should render children and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrdersTimeSeries
    dimension={Object {}}
    dimensions={
      Object {
        "ordersByBranch": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByDate": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByDeliveryArea": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByPaymentMethod": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersBySize": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByTimeOfDay": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByWeekDay": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
      }
    }
    onFilter={[MockFunction]}
    orders={
      Object {
        "add": [Function],
        "all": [Function],
        "allFiltered": [Function],
        "dimension": [Function],
        "groupAll": [Function],
        "isElementFiltered": [Function],
        "onChange": [Function],
        "remove": [Function],
        "size": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FlexGridRow
        gutter={48}
        justify="space-around"
      >
        <TimeSeries
          data={
            Array [
              Object {
                "key": "01-02-2017",
                "value": 1,
              },
              Object {
                "key": "02-06-2018",
                "value": 1,
              },
              Object {
                "key": "04-05-2015",
                "value": 1,
              },
              Object {
                "key": "04-12-2014",
                "value": 1,
              },
              Object {
                "key": "05-08-2017",
                "value": 1,
              },
              Object {
                "key": "05-10-2017",
                "value": 1,
              },
              Object {
                "key": "06-12-2015",
                "value": 1,
              },
              Object {
                "key": "09-07-2015",
                "value": 1,
              },
              Object {
                "key": "10-02-2016",
                "value": 1,
              },
              Object {
                "key": "10-08-2016",
                "value": 1,
              },
              Object {
                "key": "13-04-2017",
                "value": 1,
              },
              Object {
                "key": "16-12-2015",
                "value": 1,
              },
              Object {
                "key": "17-09-2016",
                "value": 1,
              },
              Object {
                "key": "21-06-2018",
                "value": 1,
              },
              Object {
                "key": "22-05-2018",
                "value": 1,
              },
              Object {
                "key": "23-06-2018",
                "value": 1,
              },
              Object {
                "key": "24-02-2016",
                "value": 1,
              },
              Object {
                "key": "24-07-2014",
                "value": 1,
              },
              Object {
                "key": "24-10-2017",
                "value": 1,
              },
              Object {
                "key": "25-12-2015",
                "value": 1,
              },
              Object {
                "key": "26-06-2014",
                "value": 1,
              },
              Object {
                "key": "30-07-2017",
                "value": 1,
              },
              Object {
                "key": "30-09-2017",
                "value": 1,
              },
              Object {
                "key": "31-03-2018",
                "value": 1,
              },
              Object {
                "key": "31-05-2017",
                "value": 1,
              },
              Object {
                "key": "31-08-2017",
                "value": 1,
              },
              Object {
                "key": "01-01-2017",
                "value": 1,
              },
            ]
          }
          maxPoints={150}
          x="key"
          y="value"
        />
      </FlexGridRow>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <TimeSeries
          data={
            Array [
              Object {
                "key": "01-02-2017",
                "value": 1,
              },
              Object {
                "key": "02-06-2018",
                "value": 1,
              },
              Object {
                "key": "04-05-2015",
                "value": 1,
              },
              Object {
                "key": "04-12-2014",
                "value": 1,
              },
              Object {
                "key": "05-08-2017",
                "value": 1,
              },
              Object {
                "key": "05-10-2017",
                "value": 1,
              },
              Object {
                "key": "06-12-2015",
                "value": 1,
              },
              Object {
                "key": "09-07-2015",
                "value": 1,
              },
              Object {
                "key": "10-02-2016",
                "value": 1,
              },
              Object {
                "key": "10-08-2016",
                "value": 1,
              },
              Object {
                "key": "13-04-2017",
                "value": 1,
              },
              Object {
                "key": "16-12-2015",
                "value": 1,
              },
              Object {
                "key": "17-09-2016",
                "value": 1,
              },
              Object {
                "key": "21-06-2018",
                "value": 1,
              },
              Object {
                "key": "22-05-2018",
                "value": 1,
              },
              Object {
                "key": "23-06-2018",
                "value": 1,
              },
              Object {
                "key": "24-02-2016",
                "value": 1,
              },
              Object {
                "key": "24-07-2014",
                "value": 1,
              },
              Object {
                "key": "24-10-2017",
                "value": 1,
              },
              Object {
                "key": "25-12-2015",
                "value": 1,
              },
              Object {
                "key": "26-06-2014",
                "value": 1,
              },
              Object {
                "key": "30-07-2017",
                "value": 1,
              },
              Object {
                "key": "30-09-2017",
                "value": 1,
              },
              Object {
                "key": "31-03-2018",
                "value": 1,
              },
              Object {
                "key": "31-05-2017",
                "value": 1,
              },
              Object {
                "key": "31-08-2017",
                "value": 1,
              },
              Object {
                "key": "01-01-2017",
                "value": 1,
              },
            ]
          }
          maxPoints={150}
          x="key"
          y="value"
        />,
        "gutter": 48,
        "justify": "space-around",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "key": "01-02-2017",
              "value": 1,
            },
            Object {
              "key": "02-06-2018",
              "value": 1,
            },
            Object {
              "key": "04-05-2015",
              "value": 1,
            },
            Object {
              "key": "04-12-2014",
              "value": 1,
            },
            Object {
              "key": "05-08-2017",
              "value": 1,
            },
            Object {
              "key": "05-10-2017",
              "value": 1,
            },
            Object {
              "key": "06-12-2015",
              "value": 1,
            },
            Object {
              "key": "09-07-2015",
              "value": 1,
            },
            Object {
              "key": "10-02-2016",
              "value": 1,
            },
            Object {
              "key": "10-08-2016",
              "value": 1,
            },
            Object {
              "key": "13-04-2017",
              "value": 1,
            },
            Object {
              "key": "16-12-2015",
              "value": 1,
            },
            Object {
              "key": "17-09-2016",
              "value": 1,
            },
            Object {
              "key": "21-06-2018",
              "value": 1,
            },
            Object {
              "key": "22-05-2018",
              "value": 1,
            },
            Object {
              "key": "23-06-2018",
              "value": 1,
            },
            Object {
              "key": "24-02-2016",
              "value": 1,
            },
            Object {
              "key": "24-07-2014",
              "value": 1,
            },
            Object {
              "key": "24-10-2017",
              "value": 1,
            },
            Object {
              "key": "25-12-2015",
              "value": 1,
            },
            Object {
              "key": "26-06-2014",
              "value": 1,
            },
            Object {
              "key": "30-07-2017",
              "value": 1,
            },
            Object {
              "key": "30-09-2017",
              "value": 1,
            },
            Object {
              "key": "31-03-2018",
              "value": 1,
            },
            Object {
              "key": "31-05-2017",
              "value": 1,
            },
            Object {
              "key": "31-08-2017",
              "value": 1,
            },
            Object {
              "key": "01-01-2017",
              "value": 1,
            },
          ],
          "maxPoints": 150,
          "x": "key",
          "y": "value",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FlexGridRow
          gutter={48}
          justify="space-around"
        >
          <TimeSeries
            data={
              Array [
                Object {
                  "key": "01-02-2017",
                  "value": 1,
                },
                Object {
                  "key": "02-06-2018",
                  "value": 1,
                },
                Object {
                  "key": "04-05-2015",
                  "value": 1,
                },
                Object {
                  "key": "04-12-2014",
                  "value": 1,
                },
                Object {
                  "key": "05-08-2017",
                  "value": 1,
                },
                Object {
                  "key": "05-10-2017",
                  "value": 1,
                },
                Object {
                  "key": "06-12-2015",
                  "value": 1,
                },
                Object {
                  "key": "09-07-2015",
                  "value": 1,
                },
                Object {
                  "key": "10-02-2016",
                  "value": 1,
                },
                Object {
                  "key": "10-08-2016",
                  "value": 1,
                },
                Object {
                  "key": "13-04-2017",
                  "value": 1,
                },
                Object {
                  "key": "16-12-2015",
                  "value": 1,
                },
                Object {
                  "key": "17-09-2016",
                  "value": 1,
                },
                Object {
                  "key": "21-06-2018",
                  "value": 1,
                },
                Object {
                  "key": "22-05-2018",
                  "value": 1,
                },
                Object {
                  "key": "23-06-2018",
                  "value": 1,
                },
                Object {
                  "key": "24-02-2016",
                  "value": 1,
                },
                Object {
                  "key": "24-07-2014",
                  "value": 1,
                },
                Object {
                  "key": "24-10-2017",
                  "value": 1,
                },
                Object {
                  "key": "25-12-2015",
                  "value": 1,
                },
                Object {
                  "key": "26-06-2014",
                  "value": 1,
                },
                Object {
                  "key": "30-07-2017",
                  "value": 1,
                },
                Object {
                  "key": "30-09-2017",
                  "value": 1,
                },
                Object {
                  "key": "31-03-2018",
                  "value": 1,
                },
                Object {
                  "key": "31-05-2017",
                  "value": 1,
                },
                Object {
                  "key": "31-08-2017",
                  "value": 1,
                },
                Object {
                  "key": "01-01-2017",
                  "value": 1,
                },
              ]
            }
            maxPoints={150}
            x="key"
            y="value"
          />
        </FlexGridRow>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <TimeSeries
            data={
              Array [
                Object {
                  "key": "01-02-2017",
                  "value": 1,
                },
                Object {
                  "key": "02-06-2018",
                  "value": 1,
                },
                Object {
                  "key": "04-05-2015",
                  "value": 1,
                },
                Object {
                  "key": "04-12-2014",
                  "value": 1,
                },
                Object {
                  "key": "05-08-2017",
                  "value": 1,
                },
                Object {
                  "key": "05-10-2017",
                  "value": 1,
                },
                Object {
                  "key": "06-12-2015",
                  "value": 1,
                },
                Object {
                  "key": "09-07-2015",
                  "value": 1,
                },
                Object {
                  "key": "10-02-2016",
                  "value": 1,
                },
                Object {
                  "key": "10-08-2016",
                  "value": 1,
                },
                Object {
                  "key": "13-04-2017",
                  "value": 1,
                },
                Object {
                  "key": "16-12-2015",
                  "value": 1,
                },
                Object {
                  "key": "17-09-2016",
                  "value": 1,
                },
                Object {
                  "key": "21-06-2018",
                  "value": 1,
                },
                Object {
                  "key": "22-05-2018",
                  "value": 1,
                },
                Object {
                  "key": "23-06-2018",
                  "value": 1,
                },
                Object {
                  "key": "24-02-2016",
                  "value": 1,
                },
                Object {
                  "key": "24-07-2014",
                  "value": 1,
                },
                Object {
                  "key": "24-10-2017",
                  "value": 1,
                },
                Object {
                  "key": "25-12-2015",
                  "value": 1,
                },
                Object {
                  "key": "26-06-2014",
                  "value": 1,
                },
                Object {
                  "key": "30-07-2017",
                  "value": 1,
                },
                Object {
                  "key": "30-09-2017",
                  "value": 1,
                },
                Object {
                  "key": "31-03-2018",
                  "value": 1,
                },
                Object {
                  "key": "31-05-2017",
                  "value": 1,
                },
                Object {
                  "key": "31-08-2017",
                  "value": 1,
                },
                Object {
                  "key": "01-01-2017",
                  "value": 1,
                },
              ]
            }
            maxPoints={150}
            x="key"
            y="value"
          />,
          "gutter": 48,
          "justify": "space-around",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "key": "01-02-2017",
                "value": 1,
              },
              Object {
                "key": "02-06-2018",
                "value": 1,
              },
              Object {
                "key": "04-05-2015",
                "value": 1,
              },
              Object {
                "key": "04-12-2014",
                "value": 1,
              },
              Object {
                "key": "05-08-2017",
                "value": 1,
              },
              Object {
                "key": "05-10-2017",
                "value": 1,
              },
              Object {
                "key": "06-12-2015",
                "value": 1,
              },
              Object {
                "key": "09-07-2015",
                "value": 1,
              },
              Object {
                "key": "10-02-2016",
                "value": 1,
              },
              Object {
                "key": "10-08-2016",
                "value": 1,
              },
              Object {
                "key": "13-04-2017",
                "value": 1,
              },
              Object {
                "key": "16-12-2015",
                "value": 1,
              },
              Object {
                "key": "17-09-2016",
                "value": 1,
              },
              Object {
                "key": "21-06-2018",
                "value": 1,
              },
              Object {
                "key": "22-05-2018",
                "value": 1,
              },
              Object {
                "key": "23-06-2018",
                "value": 1,
              },
              Object {
                "key": "24-02-2016",
                "value": 1,
              },
              Object {
                "key": "24-07-2014",
                "value": 1,
              },
              Object {
                "key": "24-10-2017",
                "value": 1,
              },
              Object {
                "key": "25-12-2015",
                "value": 1,
              },
              Object {
                "key": "26-06-2014",
                "value": 1,
              },
              Object {
                "key": "30-07-2017",
                "value": 1,
              },
              Object {
                "key": "30-09-2017",
                "value": 1,
              },
              Object {
                "key": "31-03-2018",
                "value": 1,
              },
              Object {
                "key": "31-05-2017",
                "value": 1,
              },
              Object {
                "key": "31-08-2017",
                "value": 1,
              },
              Object {
                "key": "01-01-2017",
                "value": 1,
              },
            ],
            "maxPoints": 150,
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
